src/gov/nist/core/LexerCore.java

```diff
@@ -736,7 +736,8 @@ public String number() throws ParseException {
 
         int startIdx = ptr;
         try {
-            if (!isDigit(lookAhead(0))) {
+            char c = lookAhead(0);
+            if (c != '-' && !isDigit(c)) {
```

Comment:
my concern is that this requirement seems to be specific for Content-Length header, but this piece of code we are modifying is applied to all sort of numberic header conversions.., hence im afraid we can break contract for other headers...